x-pack/plugins/ml/public/application/supplied_configurations/supplied_configurations_flyout/overview_tab_content.tsx

```diff
@@ -0,0 +1,345 @@
+  EuiDescriptionListDescription,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiNotificationBadge,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { asyncForEach } from '@kbn/std';
+import type { Module } from '../../../../common/types/modules';
+import { useMlKibana } from '../../contexts/kibana';
+import type { TabIdType, KibanaAssetType } from './flyout';
+import { TAB_IDS } from './flyout';
+import { DataViewsTable } from './data_views_table';
+import { isLogoObject } from '../supplied_configurations';
+
+export const LABELS = {
+  dashboard: (
+    <FormattedMessage
+      id="xpack.ml.anomalyDetection.suppliedConfigurationsFlyout.dashboardLabel"
+      defaultMessage="Dashboard"
+    />
+  ),
+  jobs: (
+    <FormattedMessage
+      id="xpack.ml.anomalyDetection.suppliedConfigurationsFlyout.jobsLabel"
+      defaultMessage="Jobs"
+    />
+  ),
+  search: (
+    <FormattedMessage
+      id="xpack.ml.anomalyDetection.suppliedConfigurationsFlyout.searchLabel"
+      defaultMessage="Search"
+    />
+  ),
+  visualization: (
+    <FormattedMessage
+      id="xpack.ml.anomalyDetection.suppliedConfigurationsFlyout.visualizationLabel"
+      defaultMessage="Visualization"
+    />
+  ),
+};
+
+export type LabelId = keyof typeof LABELS;
+
+export interface DataViewInfo {
+  id: string;
+  title: string;
+}
+
+const ListDescriptionItem = ({
+  label,
+  onClick,
+  value,
+}: {
+  label?: LabelId;
+  value: number | string;
+  onClick?: () => void;
+}) => (
+  <>
+    <EuiDescriptionListDescription>
+      {label ? (
+        <EuiButtonEmpty color={'text'} onClick={onClick}>
+          {LABELS[label]}
+        </EuiButtonEmpty>
+      ) : (
+        ''
+      )}
+      {label ? ' ' : ''}
+      <EuiNotificationBadge size={label ? 's' : 'm'} color="subdued">
+        {value}
+      </EuiNotificationBadge>
+    </EuiDescriptionListDescription>
+    {label ? null : <EuiSpacer size="s" />}
+  </>
+);
+
+interface Props {
+  module: Module;
+  setSelectedTabId: (tabId: TabIdType) => void;
+  setSelectedKibanaSubTab: (kibanaSubTab: KibanaAssetType) => void;
+}
+
+export const OverviewTabContent: FC<Props> = ({
+  module,
+  setSelectedTabId,
+  setSelectedKibanaSubTab,
+}) => {
+  const [runningDataRecognizer, setRunningDataRecognizer] = useState<boolean>(false);
+  const [recognizerWasRun, setRecognizerWasRun] = useState<boolean>(false);
+  const [matchingDataViews, setMatchingDataViews] = useState<DataViewInfo[]>([]);
+  const {
+    services: {
+      docLinks,
+      mlServices: {
+        mlApiServices: { recognizeModule },
+      },
+      data: { dataViews: dataViewsService },
+    },
+  } = useMlKibana();
+  const logsConfigsUrl = docLinks.links.ml.logsAnomalyDetectionConfigs;
+  const metricsConfigsUrl = docLinks.links.ml.metricsAnomalyDetectionConfigs;
+
+  const runDataRecongizer = async () => {
+    setRunningDataRecognizer(true);
+    const result = await recognizeModule({ moduleId: module.id });
```

Comment:
This function fires off a lot of requests, one for every data view which could be a lot.
These calls to the `dataViewsService` could be moved server side as part of the `recognize_by_module` endpoint and rather than it returning index names it could return data view details (id and name)
The client side doesn't use the index names for anything other than fetching the matching data views, so why not just make the endpoint do all that and save on the network traffic.

Reply:
Yes, that's true. I don't think there's anything available on the server side for doing something similar - grabbing the  indices associated with the data views to check which match up. I could be missing something but afaik there isn't anything comparable so that's why I did it on browser.

Could be improved for sure - maybe wrapping in a Promise all 🤔  Will dig deeper.

Reply:
>I don't think there's anything available on the server side for doing something similar - grabbing the indices associated with the data views to check which match up.

This can be done manually, load all of the index names once at the beginning and match each data view name against them. It'd save doing a request to the server for each data view. And if do it this way, it could be moved server side.

Reply:
This works, but it might be better to put it into a separate function so we can add some unit tests.

```ts
    const indices = await recognizeModule({ moduleId: module.id });
    const matching: DataViewInfo[] = [];

    if (indices?.length) {
      const dataViews = await dataViewsService.getIdsWithTitle();
      for (const { id, title } of dataViews) {
        const matches = indices.some((index) => {
          // split data view title by comma as the index pattern may contain multiple indices
          const dataviewIndexPatterns = title.split(',');
          for (const dataviewIndexPattern of dataviewIndexPatterns) {
            // replace wildcard with regex wildcard
            const regex = new RegExp(dataviewIndexPattern.replace('*', '.*'));
            if (regex.test(index)) {
              return true;
            }
          }
        });

        if (matches) {
          matching.push({ id, title });
        }
      }
    }
```

Reply:
Would this fix the issue with the data stream backed `Kibana Sample Data Logs` data view not being recognized?

Reply:
After reading Pete's comments [above](https://github.com/elastic/kibana/pull/191564#discussion_r1736022383) I now think we shouldn't be using actual indices at all because we possibly won't match data streams.

I think instead the `recognize_by_module `endpoint should load all data views and use the title to run the module's query. this will be simpler logic and we won't need to mess about with deconstructing the pattern (as my example code does)
the endpoint would then just return data view details (name and id)

Reply:
@jgowdyelastic, @peteharverson 
I was able to move all the logic to the endpoint and was able to search against the data view title (index pattern) - the endpoint now returns the matching data view id/title/name and all data view logic has been removed from the client side. Updated in 483c2bb5d94bb576142c1a37e816289b7857cacb